### ISSUE TITLE
Set up simple BIP70 Payment protocol server on multibit.org

### ISSUE THREAD

**Issue by jim618** [bitcoin-solutions/multibit-hd#491]
At the moment it is quite difficult to exercise BIP70 in real life as it's not supported by many merchants.

I suggest we set up a simple BIP70 payment protocol server on multibit.org so that:
1) Users can go to a donations page and see a BIP21 + BIP70 compliant donate link for a milli
2) They can click on it and go through the BIP70 workflow.

It would be useful for producing a 'real world' blog article too.

Payment protocol server can be based on PaymentProtocolHTTPSServer. Doesn't need any persistence and can just echo back the users comment as the PaymentACK memo.

The millis can go to the regular MultiBit donation address.

**Comment by ollekullberg** [comment]
Since BitPay supports BIP70 there are many merchants who indirectly support payment protocol, but I still like your idea, because it is a cheap way for users to test the functionality. For me it would be nice if the user can enter the amount of mBTC to donate before the actual donation. Too much work?

**Comment by gary-rowe** [comment]
Should be straightforward since we have the signing keys on the server. Perhaps the following:

1. User/developer clicks: https://multibit.org/donate.html
1. Server provides HTML5 containing form with explanatory text, QR code with BIP21 link and QR code with BIP72 link and form with custom fields over TLS
1. User clicks BIP21 link or uses QR code and can verify their wallet starts as normal
1. User clicks BIP72 link or uses QR code and can verify their wallet supports signed BIP70 `PaymentRequest`s

Page contains an Advanced section allowing customised signed `PaymentRequest` entries.

1. User enters form values and clicks submit
1. Server responds with 201 CREATED with Location (for subsequent BIP72 tests using their values, cached for 1 hour) and entity of signed `PaymentRequest` with `application/bitcoin-paymentrequest` media type - this should trigger an external media handler dialog from the browser to select the wallet

Form fields are:

`amount` - textbox - expressed in BTC as per BIP21/72
`expires` - textbox - expressed as ISO8601 in UTC (e.g. `2000-01-02T12:34:56Z`)
`memo` - textbox - the `PaymentACK` memo
`sign` - checkbox (checked) - response will be signed by `multibit.org` or not

* All outputs will point to the MultiBit donation address and thus count towards BRIT fees for that wallet
* The `PaymentACK` will respect refund addresses and use a media type of `application/bitcoin-paymentack`.
* The `PaymentACK` will contain this memo: 
```
Thank you for donating to MultiBit. 
You sent a memo to us of: <Payment.memo>
You asked for this memo in response: <form.memo>
```

**Comment by gary-rowe** [comment]
Should we pull this forward into Beta 8 to enable #483 to be properly closed?

**Comment by ollekullberg** [comment]
Hmm, I was thinking more like 

User enters form values and clicks submit

**Comment by jim618** [comment]
@gay-rowe I think it's worth shipping Beta 8 pretty much as is as it is almost done and has the Trezor 1.3.3 fix in. We know the BIP70 work wil need refining when it get used more widely - might as well get our initial support out.

**Comment by gary-rowe** [comment]
OK. We'll leave it for Release 0.1. I'll make a copy of the requirements and put them into a new issue for the website since that's where it'll end up.

@ollekullberg I'll provide two endpoints: one for simple donations through BIP21/72 and another for more technical requirements

**Comment by ollekullberg** [comment]
Sounds good

**Comment by jim618** [comment]
Do we need to do this for Release 0.1 ?
Feels like a nice-to-have rather than essential.

**Comment by gary-rowe** [comment]
Agreed. Let's bump it to 0.2 so we can get 0.1 live faster. BIP70 is going to be an evolving situation.
